app/mailers/log_email_delivery_observer.rb

```diff
@@ -1,9 +1,29 @@
 class LogEmailDeliveryObserver
   def self.delivered_email(message)
+    log_to_debugger(message)
+    link_message_with_audit(message)
+  end
+
+  def self.log_to_debugger(message)
     Rails.logger.debug <<~END_LOG_MSG
       To: #{message.to}
 
       #{message.preview.try(:body)}
     END_LOG_MSG
   end
+
+  def self.link_message_with_audit(message)
+    if message.delivery_method.is_a?(Mail::Notify::DeliveryMethod)
+      govuk_notify_id = message.delivery_method.response.id
+      audit_id = message.delivery_method.response.reference
+      audit = EmailAudit.find_by(id: audit_id)
+
+      if audit
+        audit.update!(govuk_notify_id: govuk_notify_id)
+      end
+    end
+  rescue StandardError => e
```

Comment:
I don't really have a testing strategy so would rather be on the safe side as this is a blocking call

Reply:
@asmega Can we be more specific about what you might expect here?

Reply:
I don't think so. I've not been able to do a full integration test with notify as we only have the production environment in notify. i have been using a dev key to simulate but never a live key so I don't have 100% certainty this will behave as I expect it to. I don't know what errors it could raise.

If we don't update the audit at the moment it's not really important since it's a new feature anyway and it's currently not doing that. but if it does hypothetically go wrong i'd rather the error be swallowed silently than throw an error potentially to a user if there are any synchronous calls.